lib/rouge/lexers/nial.rb

```diff
@@ -0,0 +1,167 @@
+                  "seeprimcalls", "separator", "setwidth", "settrigger",
+                  "setmessages", "setlogname", "setinterrupts",
+                  "setprompt", "setprofile", "sinh", "single",
+                  "sqrt", "solitary", "sketch", "sleep",
+                  "socket_listen", "socket_accept", "socket_close",
+                  "socket_bind", "socket_connect", "socket_getline",
+                  "socket_receive", "socket_peek", "socket_read",
+                  "socket_send", "socket_write", "solve", "split",
+                  "sortup", "string", "status", "take", "symbols",
+                  "sum", "system", "tan", "tally", "takeright",
+                  "tanh", "tell", "tr", "times", "third", "time",
+                  "toupper", "tolower", "timestamp", "tonumber",
+                  "toraw", "toplevel", "transformer", "type",
+                  "transpose", "trs", "truth", "unequal",
+                  "variable", "valence", "up", "updateall",
+                  "update", "vacate", "value", "version", "vars",
+                  "void", "watch", "watchlist", "write", "writechars",
+                  "writearray", "writefile", "writefield",
+                  "writescreen", "writerecord"]
+      end
+
+      def self.consts
+        @consts ||= Set.new %w(false null pi true)
+      end
+
+      state :root do
+        rule %r/'/, Str::Single, :str
+        rule %r/\b[lo]+\b/, Num::Bin
+        rule %r/-?\d+((\.\d*)?[eE][+-]?\d|\.)\d*/, Num::Float
+        rule %r/\-?\d+/, Num::Integer
+        rule %r/`./, Str::Char
+        rule %r/"[^\s()\[\]{}#,;]*/, Str::Symbol
+        rule %r/\?[^\s()\[\]{}#,;]*/, Generic::Error
+        rule %r/%[^;]+;/, Comment::Multiline
+        rule %r/^#(.+\n)+\n/, Comment::Multiline
+        rule %r/:=|[\{\}\[\]\(\),:;]/ do |m|
+          if self.class.punctuations.include?(m[0])
+            token Punctuation
+          else
+            token Text
+          end
+        end
+        # [".",  "!", "#", "+", "*", "-", "<<",
+        #   "/", "<", ">>", "<=", ">", "=", ">=", "@", "|", "~="]
+        rule %r'>>|>=|<=|~=|[\.!#+*\-=></|@]' do |m|
+          if self.class.operators.include?(m[0])
+            token Operator
+          else
+            token Text
+          end
+        end
+        
+        rule %r/\b[_A-Za-z]\w*\b/ do |m|
+          lower = m[0].downcase
+          if self.class.keywords.include?(lower)
+            token Keyword
+          elsif self.class.funcs.include?(lower)
+            token Keyword::Pseudo
+          elsif self.class.transformers.include?(lower)
+            token Name::Builtin
+          elsif self.class.consts.include?(lower)
+            token Keyword::Constant
+          else
+            token Name::Variable
+          end
+        end
+
+        rule %r/\s+/, Text
+      end
+      
+      state :str do
+        rule %r/''/, Str::Escape
+        rule %r/[^']+/, Str::Single
+        rule %r/'|$/, Str::Single, :pop!
+      end
+    end
+  end
+end
+
```

Comment:
**suggestion:** `linelint` does not like empty lines. Could we remove these?

Reply:
Removed the lines